flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriteRequestExecutorImpl.java

```diff
@@ -67,8 +70,11 @@ void run() {
 		} catch (Exception ex) {
 			thrown = ex;
 		} finally {
-			cleanupRequests();
-			dispatcher.fail(thrown == null ? new CancellationException() : thrown);
+			try {
+				closeAll(this::cleanupRequests, () -> dispatcher.fail(thrown == null ? new CancellationException() : thrown));
+			} catch (Exception e) {
+				LOG.error("unable to terminate properly", e);
+			}
```

Comment:
we had quite a bit of problems with error logging like that polluting the logs failing e2e tests. It would be better to add `e` as suppressed exception to the `thrown`, like `org.apache.flink.streaming.runtime.tasks.StreamTask#invoke` is doing.
```
} catch (Exception e) {
	dispatcher.fail(ExceptionUtils.firstOrSuppressed(e, thrown));
}
```
?

Reply:
I'll update the `thrown` field (`dispatcher.fail` is already called two lines above).